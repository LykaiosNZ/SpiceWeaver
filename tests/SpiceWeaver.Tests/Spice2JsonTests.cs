namespace SpiceWeaver.Tests;

[TestFixture(TestOf = typeof(Spice2Json))]
public sealed class Spice2JsonTests
{
    [Test]
    public void ConvertToJson_WhenPathIsInvalid_ShouldThrow()
    {
        var e = Assert.Throws<Spice2JsonException>(() => Spice2Json.ConvertToJson("someFakePath", TestSchema.SpiceDb));

        Snapshot.Match(e?.Message);
    }

    [Test]
    public void ConvertToJson_WhenSchemaFileIsInvalid_ShouldThrow()
    {
        var e = Assert.Throws<Spice2JsonException>(
            () => Spice2Json.ConvertToJson("spice2json", "This is not a schema."));

        Snapshot.Match(e?.Message);
    }

    [Test]
    public void ConvertToJson_ShouldGenerateExpectedOutput()
    {
        var result = Spice2Json.ConvertToJson("spice2json", TestSchema.SpiceDb);
        
        Snapshot.Match(result);
    }
}