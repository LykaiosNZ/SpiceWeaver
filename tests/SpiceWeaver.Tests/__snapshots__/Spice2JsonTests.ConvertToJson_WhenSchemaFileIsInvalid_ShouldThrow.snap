spice2json exited with code 1. Output: parse error in ``, line 1, column 1: Unexpected token at root level: TokenTypeIdentifier
